Objects/abstract.c

```diff
@@ -495,6 +500,30 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+int
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *result;
+    int size;
+
+    structmodule = PyImport_ImportModule("struct");
```

Comment:
it's a strong reference: you must Py_DECREF(structmodule) to prevent a reference leak. I suggest to add a "done:" label which does DECREF and replace return NULL with "goto done".